Sources/Datadog/Core/System/LaunchTimeProvider.swift

```diff
@@ -16,10 +16,7 @@ internal protocol LaunchTimeProviderType {
 
 internal class LaunchTimeProvider: LaunchTimeProviderType {
     var launchTime: TimeInterval? {
-        // Following dynamic dispatch synchronisation mutes TSAN issues when running tests from CLI.
-        objc_sync_enter(self)
```

Comment:
I couldn't reproduce the TSAN issues locally, but I believe that if we don't involve any Objc objects (and call a C function) in that call paths there can't be any dynamic dispatches involved to begin with.

Reply:
Sounds reasonable 👍. My suspicion behind this TSAN issue was indeed around the dynamic dispatch with this call.

Reply:
I actually had to come back to using `objc_sync_enter()`/`objc_sync_exit()`. I could not reproduce locally the test failure, but based on Bitrise logs, TSAN reported a potential data race on the call to `AppLaunchTime()` where a write happened according to TSAN. When reporting, TSAN crashes on a `SIGABRT`.

The only write in that C function might come from `pthread_rwlock_rdlock(&rwLock);` which in itself must be thread safe (otherwise nothing can be). So anyway, this is back here, it might be extraneous but let's trust TSAN for now.